### ISSUE TITLE
Citus CI failing

### ISSUE THREAD

**Issue by mrigger** [sqlancer/sqlancer#698]
The Citus CI has been failing for a while now. For example, https://github.com/sqlancer/sqlancer/actions/runs/3956119066/jobs/6776946798 shows failures as outlined below.

```
CREATE DATABASE database2 ; --- ERROR: database "database2" is being accessed by other users
``` 

```sql
INSERT INTO t1(c0) VALUES(B'1111111111111111111111111111111111101000111000001001100000100110'), (B'1111111111111111111111111111111111010111011111001111111000100000'), (B'1111111111111111111111111111111110001100100110000101001001100011') ON CONFLICT  DO NOTHING; --- ERROR: columnar_tuple_insert_speculative not implemented
```

**Comment by mrigger** [comment]
Previously reported an issue at https://github.com/citusdata/citus/issues/6670, which might have fixed this. cc @malwaregarry

**Comment by malwaregarry** [comment]
Still showing up on v11.3  and v12.1. 

Perhaps we can add the error to ExpectedError or do  some rate limiting?

**Comment by mrigger** [comment]
> Perhaps we can add the error to ExpectedError

I think this would just mask the issue. I think it would be better to try addressing it.

CC'ing @onurctirtir and @gokhangulbiz, since they worked on the issue/fix (see https://github.com/citusdata/citus/issues/6670).

**Comment by malwaregarry** [comment]
resolved in #960 by disconnecting clients first
